File: AndroidEdu/BounceBall/app/src/main/java/edu/ucsb/cs/cs190i/xuanwang/bounceball/MainActivity.java
/*
 *  Copyright (c) 2017 - present, Xuan Wang
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

package edu.ucsb.cs.cs190i.xuanwang.bounceball;

import android.graphics.Color;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    View bouncingBallView = new BouncingBallView(this);
    bouncingBallView.setBackgroundColor(Color.BLACK);

    setContentView(bouncingBallView);

    DrawingThread drawingThread = new DrawingThread(bouncingBallView, 50);
    drawingThread.start();
  }

}

File: AndroidEdu/KotlinPlygrd/app/src/main/java/edu/ucsb/cs/cs190i/xuanwang/kotlinplayground/MainActivity.kt
/*
 *  Copyright (c) 2017 - present, Xuan Wang
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

package edu.ucsb.cs.cs190i.xuanwang.kotlinplayground

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import org.jetbrains.anko.*
import org.jetbrains.anko.sdk25.coroutines.onClick

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    verticalLayout {
      val t = textView{
        text = "hh"
      }
      button("change text") {
        onClick { t.text = "nonono" }
      }.lparams(width = matchParent) {
        horizontalMargin = dip(5)
        topMargin = dip(10)
      }
    }
  }
}

